{
  try {
    syncFuture.get();
  }
 catch (  InterruptedException ie) {
    Thread.currentThread().interrupt();
    IOException ioe=new InterruptedIOException();
    ioe.initCause(ie);
    throw ioe;
  }
catch (  ExecutionException e) {
    throw ensureIOException(e.getCause());
  }
}
